GVFS/GVFS.Service/Mac/GVFSService.cs

```diff
@@ -0,0 +1,122 @@
+
+        public GVFSService(JsonTracer tracer)
+        {
+            this.tracer = tracer;
+            this.serviceName = GVFSConstants.Service.ServiceName;
+        }
+
+        public void RunWithArgs(string[] args)
+        {
+            if (this.serviceThread != null)
+            {
+                throw new InvalidOperationException("Cannot start service twice in a row.");
+            }
+
+            // TODO: 865304 Used for functional tests and development only. Replace with a smarter appConfig-based solution
+            string serviceName = args.FirstOrDefault(arg => arg.StartsWith(ServiceNameArgPrefix));
+            if (serviceName != null)
+            {
+                this.serviceName = serviceName.Substring(ServiceNameArgPrefix.Length);
+            }
+
+            try
+            {
+                this.serviceDataLocation = Paths.GetServiceDataRoot(this.serviceName);
+                this.StartServiceThread();
+                this.serviceThread.Join();
+            }
+            catch (Exception e)
+            {
+                this.LogExceptionAndExit(e, nameof(this.RunWithArgs));
+            }
+        }
+
+        private void StartServiceThread()
+        {
+            if (this.serviceStopped != null)
+            {
+                return;
+            }
+
+            this.serviceStopped = new ManualResetEvent(false);
+            this.serviceThread = new Thread(this.ServiceThreadMain);
+
+            this.serviceThread.Start();
+        }
+
+        private void ServiceThreadMain()
+        {
+            try
+            {
+                EventMetadata metadata = new EventMetadata();
+                metadata.Add("Version", ProcessHelper.GetCurrentProcessVersion());
+                this.tracer.RelatedEvent(EventLevel.Informational, $"{nameof(GVFSService)}_{nameof(this.ServiceThreadMain)}", metadata);
+
+                string pipeName = this.serviceName + ".Pipe";
+                this.tracer.RelatedInfo("Starting pipe server with name: " + pipeName);
+
+                using (NamedPipeServer pipeServer = NamedPipeServer.StartNewServer(pipeName, this.tracer, this.HandleRequest))
+                {
+                    this.serviceStopped.WaitOne();
+                }
+            }
+            catch (Exception e)
+            {
+                this.LogExceptionAndExit(e, nameof(this.ServiceThreadMain));
+            }
+        }
+
+        private void HandleRequest(ITracer tracer, string request, NamedPipeServer.Connection connection)
+        {
+            NamedPipeMessages.Message message = NamedPipeMessages.Message.FromString(request);
+            if (string.IsNullOrWhiteSpace(message.Header))
+            {
+                return;
+            }
+
+            using (ITracer activity = this.tracer.StartActivity(message.Header, EventLevel.Informational, new EventMetadata { { "request", request } }))
+            {
+                switch (message.Header)
+                {

```

Comment:
Ignores all messages for the time being

Reply:
Do we expect any messages at this point?

Reply:
No. Removed it.